include/linker/linker-defs.h

```diff
@@ -291,6 +291,15 @@ extern char z_user_stacks_start[];
 extern char z_user_stacks_end[];
 #endif /* CONFIG_USERSPACE */
 
+#ifdef CONFIG_THREAD_LOCAL_STORAGE
+extern char __tdata_start[];
+extern char __tdata_end[];
+extern char __tbss_start[];
+extern char __tbss_end[];
+extern char __tls_start[];
+extern char __tls_end[];
+#endif /* CONFIG_THREAD_LOCAL_STORAGE */
+
 #endif /* ! _ASMLANGUAGE */
```

Comment:
I've found it very useful to have some size symbols as well; this avoids doing subtracts at run-time. TLS will also need the address of where the initial values for .tdata live in ROM so they can be copied into a new TLS segment when that is created.

picolibc uses the following symbols:

```
extern char __tdata_source[];	/* Source of TLS initialization data (in ROM) */
extern char __tdata_size[];	/* Size of TLS initized data */
extern char __tbss_size[];	/* Size of TLS zero-filled data */
extern char __tls_size[];	/* Total size of TLS area */
```

Reply:
Added the size symbols.